### PR TITLE
Convert  TeacherSections component to functional component

### PR DESCRIPTION
**Pull request by mgc1194**
As part of the Colorado Privacy Act work the Platform team does, we add warning banners to different `Sections` components.

[P20-815](https://codedotorg.atlassian.net/browse/P20-815) requires adding a Warning Notification to the [TeacherSections](https://github.com/code-dot-org/code-dot-org/blob/50a3e70c3295eb21962c71f119167460b4e6ea39/apps/src/templates/studioHomepages/TeacherSections.jsx) component in the [TeacherHomepage](https://github.com/code-dot-org/code-dot-org/blob/98c6fa7aff8724aad45e34c75204d3c2e3b3ff7a/apps/src/templates/studioHomepages/TeacherHomepage.jsx#L270).

I'd like to update the existing code and convert the class component into a functional component.

## Links

- Jira task: [P20-815](https://codedotorg.atlassian.net/browse/P20-815)

## Testing story

Screenshot 2024-04-19 at 18 34 20

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/studioHomepages/TeacherSections.jsx [hunk 1]
```diff
@@ -22,81 +22,82 @@ import {
...
-    );
-  }
+function TeacherSections({
+  asyncLoadSectionData,
+  asyncLoadCoteacherInvite,
+  coteacherInvite,
+  coteacherInviteForPl,
+  studentSectionIds,
+  plSectionIds,
+  hiddenPlSectionIds,
```
Comment (mgc1194) [inline apps/src/templates/studioHomepages/TeacherSections.jsx]: I don't see this prop being used anywhere in this component. is it some legacy prop that we could get rid of, or is it missing from `shouldRenderPlSections`?

**Review by stephenliang — APPROVED** [review]

File: apps/src/templates/studioHomepages/TeacherSections.jsx (hunk 1 above)
Comment (Nokondi) [reply apps/src/templates/studioHomepages/TeacherSections.jsx]: Not sure about this - I'll ask in the teacher tools channel

Comment (Nokondi) [reply apps/src/templates/studioHomepages/TeacherSections.jsx]: https://codedotorg.slack.com/archives/C045UAX4WKH/p1714656205422699

**Review by Nokondi — APPROVED** [review]
Looks great! Thanks for updating this!

**Pull request merged**